File: untold_comparison_programs/08_oop/go/main.go
package main

import "fmt"

type Person struct {
    Name string
    Age  int
}

func (p *Person) greet() {
    fmt.Printf("Hi, I am %s and I'm %d years old!\n", p.Name, p.Age)
}

type Developer struct {
    Person
    Language string
}

func (d *Developer) introduce() {
    fmt.Printf("I code in %s\n", d.Language)
}

func main() {
    // Create person object
    p := Person{Name: "Untold", Age: 2}
    p.greet()

    // Modify properties
    p.Age = 3
    fmt.Printf("Updated age: %d\n", p.Age)

    // Use inherited behavior
    dev := Developer{
        Person:   Person{Name: "Dev", Age: 25},
        Language: "Go",
    }
    dev.greet()
    dev.introduce()
}

File: untold_comparison_programs/08_oop/python/main.py
# Classes and Objects (OOP)
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def greet(self):
        print(f"Hi, I am {self.name} and I'm {self.age} years old!")

class Developer(Person):
    def __init__(self, name, age, language):
        super().__init__(name, age)
        self.language = language

    def introduce(self):
        print(f"I code in {self.language}")

# Create person object
p = Person("Untold", 2)
p.greet()

# Modify properties
p.age = 3
print(f"Updated age: {p.age}")

# Use inherited behavior
dev = Developer("Dev", 25, "Python")
dev.greet()
dev.introduce()